---
title: "Phylogenetic scale of water mite specialization in the order Odonata"
author: "Jacob K. Harvey"
date: "08/03/2021"
output:
  html_document: default
  word_document: default
bibliography: bibliography.json
---

```{css, echo=FALSE}
p{
  text-align: justify;
}
.caption{
  font-weight: lighter;
  font-size: smaller;
}
h1 {
  font-size: larger;
}
h2{
  font-size: medium;
}

```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(results = FALSE)
```
# 1	Introduction

Specialization, the process by which organisms limit themselves to a subset of 
the available resources, is an important unifying concept in ecology 
[@poisot2011]. Every organism needs some sort of resource to survive and as such
can be placed somewhere on the spectrum of specialization. Specialization has 
many important ecological consequences, such as enabling the coexistence of 
species through resource partitioning [@poisot2011].
**[maintenance of biodiversity, chaotic patch dynamics, population control etc.]**

Researchers have found many host traits which seem to determine the degree to
which their parasites specialize. One such pattern which is thought to be robust
is that host species with unstable population dynamics tend to have more
generalist parasites as specialist parasites would be at a very high risk of
extinction when the population dips too low [@bush1994]. As such species which
occur in low abundance should host less specialist parasites. We might also
expect this relationship to hold with mass as well with smaller organisms
hosting more specialist mites as small organisms tend to be more abundant than
large organisms [@lewis2008]. 

The argument has also been made for the opposite relationship, that large 
organisms should host more specialist parasites. The argument is that they are
longer lived and serve as more reliable resources [@poulin2007a]. This argument
might not hold in general, especially when dealing with non-vertebrate hosts 
such as insects which are short-lived in general.

Another pattern is that very speciose lineages should host more generalist 
parasites [@poulin2007a]. This is because parasites may often be less successful
on species the more distantly they are related to their current hosts 
[@perlman2003; @krasnov2004; @poulin2011]. Many closely related species creates
somewhat of a bridge, allowing parasites to easily adapt to new host species and
expand their host range. 

Finally, host defences should have an effect on the degree to which their 
parasites specialize. Weak host defences should allow parasites to be successful
on a host with only minimal adaptation therefore allowing success on a wide 
range of hosts. @moller2005 found that, in birds, species with weak immune 
responses host more generalist fleas, than birds with strong immune systems; 
this pattern was also observed for flatworms which parasitize freshwater fishes
[@locke2010].

A very important topic in parasite ecology for us is the debate between the
resource breadth hypothesis and the trade-off hypothesis. Each attempts to
explain based on a parasites degree of host specialization, what its abundance
should be. The resource breadth hypothesis posits that parasites which have
the widest host range (generalists) should be the most abundant, based on the
logic that adaptations that make them efficient parasites on one species also
make them able to infect a wide range of hosts [@krasnov2004a]. The trade-off 
hypothesis on the other hand, posits a trade-off; close adaptation to a host, 
which allows a parasite to reach very high abundance on that host, come at the 
expense of being less adapted to other hosts and therefore reaching lower 
abundances on them [@poulin1998]. 

Many parasite-host systems were examined for evidence towards one of these two 
hypothesis and many have found evidence for resource breadth [@krasnov2004a]; 
many also have found evidence for the trade-off hypothesis [@poulin1998; 
@poulin2004]; and still others have found no clear evidence for either 
[@morand2000]. Faced with these conflicting results, authors have tried to 
explain these different outcomes with the phylogenetic structure of the host 
community starting with @krasnov2004a and later refined by @pinheiro2016 in 
their *integrative hypothesis of parasite specialization*. This posits that 
both the trade-off and the resource-breadth effects are always in effect but 
manifest at different phylogenetic scales. Adaptations that help a parasite 
reach high abundance on a host also help it reach high abundances on other 
hosts, but only those which are closely related, the effect diminishes the 
more distantly related the hosts. Since the phylogentic distance between a 
group of hosts do not vary smoothly but have abrupt jumps corresponding to 
deep branches in the phylogeny, we can see both the niche-breadth and 
trade-off patterns depending on how the study system is circumscribed.

Many organisms, and especially parasites, will exhibit strong phylogenetic 
signal in the resources they use. Their resource use is not statistically 
independent of the phylogeny; resources are not distributed randomly with
respect to the phylogeny. They utilize a set of related hosts, and will
often be less successful on hosts the more distant the relation to their primary
host [@perlman2003; @poulin2011; @krasnov2004]. Specialization can simply be 
expressed as the number of hosts a parasite exploits, but the phylogenetic 
relationships between the hosts adds another dimension. A parasite exploiting 
two closely related hosts is more specialist than a parasite exploiting two 
distantly related hosts. To address this, indices of the phylogenetic diversity
are widely used to express degree of specialization accounting for the 
phylogentic similarity of species [@poulin2011]. Non-phylogenetic indices can
also be used if corrected for non-independence [@poisot2012]. 

Phylogenetic scale also becomes an important consideration when considering 
parasites whose host choice is phylogenetically determined (see @graham2018). 
In this use case, it is conceptually very similar to phylogenetic diversity 
indices. It expresses how narrow or how broad of a collection of related hosts 
are. Phylogenetic scale can be a much more intuitive way of expressing this 
than diversity indices. Indices often are scaled to a number between zero and 
one, or otherwise simply don't have values which have intuitive meaning. A 
measure of phylogenetic scale can take on natural units which have an intuitive
link to the phylogeny, such as an expression how many millions of years of 
evolutionary history explain the current host community of a parasite. This also
makes comparisons between different studies easier to make, since the values can
be on a common scale of evolutionary time span.

Given the relative simplicity of studying host-parasite relationships, there 
has been much work in this area, though mostly on vertebrate hosts. Much less 
work has been done on insect hosts [@mlynarek2015]. Water-mite (Hydrachnidia) 
parasitism on odonates provides an ideal model system for the study of 
parasitism on insect hosts because of odonates’ widespread range, high 
abundance, and relative ease of identification. While there has 
been research on the intensity of water-mite parasitism, much less is known 
about how specialized these mites are and what lineages of odonates host the 
most specialized parasites [@loscerbo2020; @mlynarek2015]. This might be due to
the difficulty in identifying water-mites to the species level given their lack
of morphological variation.

Odonates have single terrestrial adult stage, early instars, called nymphs are
all aquatic. Newly hatched Arenurrus water mite larvae find final instar nymphs
in the water column and attach themselves. When the nymph is ready to moult, it
climbs from the water, normally onto some vegetation, its exoskeleton splits
down the back and the teneral adult emerges. Before the cuticle has had the
chance to harden, the mites transfer from the exuviae onto the teneral Odonate.
It then pierces the cuticle with its mouthparts and begins to engorge itself.
The engorgement process may take a few days to complete, upon which they detach
themselves while the odonate is over water where they metamorphose into an 
adult, predatory, water mite. Another genus of water mite, Limnochares, is also
known to parasitize odonates, but it uses a different strategy. The larval mite
crawls onto some vegetation on which it waits for an Odonate to perch. When an
odonate comes near enough it climbs on and pierces its cuticle and begins to
engorge itself.

The order Odonata consists of two major suborders: Anisoptera and Zygoptera 
(dragonflies and damselflies) which can be contrasted in a number of ways.
anisopterans are generally larger than zygopterans and are generally present in
lower local abundance. Dragonflies also tend to have stronger defences than 
damselflies; their cuticles are thicker and they have stronger immune responses
[@blondeau].

The contrast between the two suborders would lead us to expect a difference in
selective pressures for parasites in each group. We would expect that 
dragonflies being in lower local abundance would be more likely to 
experience local extinction; this should lead to dragonflies having more 
generalist mites as the specialists would go extinct at higher rates. The higher
local abundance of damselflies should also be able to support a larger
population of mites. Host defences (cuticle thickness and immune response) being
stronger in dragonflies, we should expect them to host more specialist 
parasites [@moller2005; @locke2010]. This is in opposition to what we would
expect from the differences in abundance between dragonflies and damselflies.
Given the differences in parasite intensity between the two suborders described
by @loscerbo2020 we would expect that these abundant mites on damselflies 
represent specialists according to the integrative hypothesis of parasite 
specialization [@pinheiro2016].

Here we will reconstruct the network of odonate-mite interactions using mites 
collected from odonate specimens across Québec. To avoid the difficulty of 
identifying water-mite species, we will use COI sequencing to differentiate
mites. By inferring the phylogenetic scale at which mites specialize in this
network we will attempt to answer two hypotheses:

1. The suborder Zygoptera will host more specialist mites than Anisoptera as is
   suggested by the higher mite parasitism intensity.

2. Host abundance should be a significant predictor of the degree to which mites
   specialize on their host. Mass should also be significant given it's
   relationship with abundance.

# 2	Methods

## 2.1	Odonate sampling & Mite preparation

Specimens were collected from 83 sites (map 2.1) across Québec in the summers
of 2015, 2019, and 2020. Of these sites, specimens with mites were found at 29. 
Specimens from all years were weighed, inspected under microscope for mites, 
and had their mites counted. Field work each year was conducted according to 
a different sampling protocol. As such we can infer nothing about community 
composition or abundances in general. The odonates from 2015 were the same 
used for @arrowsmith2018 and @loscerbo2020 and were collected according to a 
standardized sampling protocol. As such, we have abundance data for only this 
subset of the data. Most species in the whole dataset are also represented in 
this subset. Since most analysis here is done at the host species level, most
of our data points can be retained when using abundance as a predictor.

Mites were removed under a dissecting scope and pooled into autoclaved 
microtubes by species and by site. Such that at each site, one sample per 
odonate species which had parasites was prepared. Specimens from 2015 were 
pooled by species only; all mites found on a particular host species at any 
site were pooled into the same sample. In our analysis, samples which were 
pooled at the site level were combined, and analysis was done only at the host
species level.

Since most odonate specimens were not collected with DNA preservation in mind,
we did not preserve all specimens in the same way. Those from 2015 were 
immersed in 95% ethanol for 12-24 hours and were then removed and allowed to 
air dry; they remained in envelopes for the next 5 years, until we removed their 
mites [@loscerbo2020]. Those from 2019 were doused with acetone, allowed to air 
dry and then stored in envelopes for a year, before we removed their mites. 
Those from 2020 were treated the same way as those from 2019 but only stored for
a few weeks. One sample from 2020 was stored immediately in 95% ethanol until 
its mites were removed.

Except for samples 1 through 5 which were a trial run, mites were stored in 95%
ethanol before DNA extraction. The longest any sample spent between mite removal
and DNA extraction was three months. To remove the ethanol in preparation for
DNA extraction the microtubes were uncovered under a fume-hood and the ethanol
was allowed to evaporate overnight.

![Map2.1](Sampling Sites.svg)
<p class = "caption">
Map 2.1 Odonate sampling sites across the Québec. (OpenStreetMaps)
</p>

## 2.2	DNA Extraction and sequencing

DNA was extracted using Qiagen’s DNEasy Blood and Tissue Kit. The extractions 
were performed according to the manual except that two 2.3 mm stainless steel
beads (BioSpec Products) were added to each tube to increase cell disruption.
During the lysis step the samples were removed from their heat bath and vortexed 
for 5 minutes at a time for a total of 30 minutes over a 3-hour long digestion. 
The first 5 samples were a trial run, these were put in a bead mill for 2 
minutes rather than using the vortex. One batch (samples 1-16) was also vortexed
for 30 minutes before being put into the heat bath. Once in the heat bath, this
batch was vortexed only occasionally as per the manufacturer's instruction for 
a few seconds at a time to agitate the material. Five negative control samples 
were also used which showed minimal cross-contamination. Samples were randomly 
assigned to batches for DNA extraction to ensure that any cross-contamination 
that occurs during the next steps is randomly distributed and would cause no 
bias to the results. All DNA samples were then stored in a -80C freezer until 
the next steps.

A 350 bp segment of Cytochrome Oxidase I was amplified using a MICOInt forward 
primer [@leray2013] and a Fold reverse primer [@arribas2016]. All samples, 
including negative controls were amplified and sequenced on an Illumina Miseq 
at the Université du Québec à Montréal CERMO-FC Genomics Platform using the 
MiSeq reagent kit v3 (2 x 300 cycles; Illumina).

## 2.3	Sequence preparation and network construction

The raw sequence data was prepared and chimeric sequences removed using DADA2 
version 1.18.0 [@callahan2016]. Both forward and reverse reads had 26 
nucleotides trimmed from the beginning of the sequence to remove primers.
Sequences were dereplicated and ASVs inferred. Forward and reverse reads were
merged with a minimum overlap of 10 nucleotides. Unless otherwise stated, the
default settings from DADA2 were used.

DADA2 was also used to assign taxonomy to all the samples. Since there is no
established reference database for assigning mite taxonomy we created our own
using all of the Arachnid and Odonate sequences stored in the BOLD database as
of September 2020 as well as BOLD's taxonomic information [@ratnasingham2007].
Any non-mite sequences were filtered out on the basis of these taxonomy 
assignments.

Rarefying read count data has been shown to throw away more data than necessary
[@mcmurdie2014]. Instead, we used a variance stabilizing transformation from the 
DESEq2 package [@anders2010]. Any ASV which still showed any abundance in a
sample after transformation was counted as a detection of an interaction. All
of the control samples showed no detections after this step except for a single
positive detection in a PCR control. We created a simple binary network since we
have too few replicates of many species to confidently place probabilistic 
weights on the interactions. Interactions in this network correspond to positive
detections of a mite ASV in a sample from an odonate species. Any mites recorded
as interacting only with a single host species were excluded as these are likely
simply do to incomplete sampling.

## 2.4	Determination of phylogenetic scale

The phylogenetic scale of a mite represents the depth in the host's phylogenetic
tree where host identity is best predicted. It takes on whatever units the
branch lengths on the tree are in. As this method required an ultrametric tree,
this is normally time. In the case of this analysis, millions of years of
evolutionary history.

This is calculated as the phylogenetically weighted Simpson's diversity of each
mite's host community [@rao1982]. When an ultrametric phylogeny is used, as we
are, this gives the mean branch length to the most recent common ancestor of any
pair of species in the community. In other words, the expected age of the most
recent common ancestor of any two randomly selected hosts.

This method is better than simply using the most recent common ancestor of all
host species as an indication of the phylogentic breadth of a parasite because
that would be very susceptible to sample contamination or to rare observations
on an incidental host. A detection of a mite interaction with a species 
distantly related to its other hosts would cause a large shift in the resultant
scale of that mite.

## 2.5 Statistical analyses

To show that different odonate lineages differ in the balance of specialist and
generalist parasites that they host, we classified each mite as either a
specialist or a generalist. Those with phylogenetic scales of greater than 100
Ma are generalist and those less than 100 Ma are specialist. In the odonate 
phylogeny this roughly corresponds to a division at approximately the family
level. This value was chosen because there is a natural gap in the distribution
of phylogenetic scales at around that value imposed by long branches in the
phylogeny.

To test how sensitive the analysis is to our choice of phylogentic scale for 
classifying mites as specialist or generalist we re-ran the models using
different values for the cutoff --  25 Ma, and 50 Ma --  to ensure all the
results remain significant.

We use Generalized Additive Models (GAMs) from the `mgcv` package [@wood2017]
to do phylogenetic regression. The phylogeny was represented using the Markov
Random Field (MRF) basis provided in the package. A MRF is a graphical model
representing the relationships between many random variables. The indented use 
they had when they were implemented in the package was for dealing with 
aggregated spatial data, such as census divisions, were neighbouring polygons
are assumed to be more similar to each other than non-adjacent polygons. This
also applies to phylogenies, were nodes which share and edge are assumed to be
more similar to nodes which are many edges distant from one another. This
relationship between the nodes is represented as a penalty matrix, penalizing
nodes toward or away from each other based on edge lengths in the phylogeny.

As a response variable we used the number of specialists and the number of
generalists on a particular odonate species with a binomial error structure.
As predictors, we used the aforementioned phylogenetic term as well as the log
transform of the average mass and abundance of an odonate host species.

# 3	Results

The phylogenetic scale analysis revealed that there are both specialist and 
generalist mites. The generalists can be found on both Anisoptera and Zygoptera, 
but specialists are only found on Zygoptera, producing a clear triangular 
pattern evident in figure 3.1.

The specialist mites on Zygoptera specialize at approximately, or a little 
above, the scale of the genera in Zygoptera. This is roughly in line with the 
work of @mlynarek2015 on Zygoptera where most of the mite OTUs they found 
strayed very little out of a single genus of hosts. 

![Figure 3.1](../figures/pscale_phylo_plot_edit.svg){width=100%}
<p class = "caption">
Figure 3.1 Phylogenetic scale of mites infesting odonates. Each point represents
an interaction between a mite and its host. As such, multiple points represent 
the same mite. Units are in millions of years. (Tree’s scale should be flipped.)
</p>

```{r}
source("scripts/specialization_gam_models.r")
source("scripts/tidying_up_regressions.r")

ps <- summary(phylo_model)
ms <- summary(mass_model)
as <- summary(abun_model)
pt <- as.data.frame(ps$s.table) 
mt <- as.data.frame(ms$p.table)
at <- as.data.frame(as$p.table)

pms <- summary(mass_phylo_model)
pmt <- as.data.frame(pms$s.table)

pas <- summary(abun_phylo_model)
pat <- as.data.frame(pas$s.table)
```

|                   |Effective df|Ref df       |Chi-squared  |p-value           |
|-------------------|:-----------|:------------|:------------|:-----------------|
|Species            |`r pt$edf`  |`r pt$Ref.df`|`r pt$Chi.sq`|$`r pt$"p-value"`$|
<p class = "caption">
Table 3.1 GAM predicting proportion of specialists with the odonate phylogeny.
R^2^_adj_ = `r ps$r.sq`
</p>

The GAM model predicting the proportion of specialists according to the odonate
phylogeny is highly significant and has strong predictive power (Table 3.1). 
Independently, both mass and abundance predict very well for the proportion of 
specialists as well (Table 3.2 & 3.3). Since mass is highly phylogenetically 
conserved in odonates, they share a very large amount of variance and it was 
impossible in our models to disentangle the two; only one predictor would be 
significant in the model depending on their order of entry despite independently
being extremely significant predictors. Abundance would also become non
significant when analyzed in the same model as the phylogenetic term.


|          |Estimate          |Standard Error        |z-value            |p-value               |
|----------|:-----------------|:---------------------|:------------------|:---------------------|
|Intercept |`r mt$Estimate[1]`|`r mt$"Std. Error"[1]`|`r mt$"z value"[1]`|$`r mt$"Pr(>|z|)"[1]`$|
|log(mass) |`r mt$Estimate[2]`|`r mt$"Std. Error"[2]`|`r mt$"z value"[2]`|$`r mt$"Pr(>|z|)"[2]`$|
<p class = "caption">
Table 3.2 GAM predicting proportion of specialists with the log of species
average mass. R^2^_adj_ = `r ms$r.sq`
</p>

|               |Estimate          |Standard Error        |z-value            |p-value               |
|---------------|:-----------------|:---------------------|:------------------|:---------------------|
|Intercept      |`r at$Estimate[1]`|`r at$"Std. Error"[1]`|`r at$"z value"[1]`|$`r at$"Pr(>|z|)"[1]`$|
|log(Abundance) |`r at$Estimate[2]`|`r at$"Std. Error"[2]`|`r at$"z value"[2]`|$`r at$"Pr(>|z|)"[2]`$|
<p class = "caption">
Table 3.3 GAM predicting proportion of specialists with abundance of the host
species. R^2^_adj_ = `r as$r.sq`
</p>

To demonstrate the strong phylogenetic conservation of mass in odonates, we fit
another GAM model predicting the mass of odonate individuals (n = `r pms$n`) 
with the odonate phylogeny. Phylogeny was, as expected, highly significant 
(p = `r pmt$"p-value"`) and was a very strong predictor of odonate mass 
(R^2^_adj_ = `r pms$r.sq`)

A similar GAM model was fit predicting host species abundance (n = `r pas$n`) 
with the host phylogeny. Phylogeny was marginally significant 
(p = `r pat$"p-value"`) but had fairly strong predictive power 
(R^2^_adj_ = `r pas$r.sq`)

```{r, out.width="100%"}
library(tidyverse)
library(cowplot)

odonates <- read_csv("datasets_derived/odonate_summaries.csv") %>%
  drop_na(spec_ratio) %>%
  filter(spec_ratio < Inf)

p1 <- ggplot(aes(x = mass, y = spec_ratio), data = odonates) +
        scale_x_log10() +
        geom_point() +
        geom_smooth(method = "gam", formula = y ~ x) + 
        xlab("Log Average Mass of Host") +
        ylab("Ratio of Specialists to Generalists") +
        theme(text = element_text(family = c("Futura Medium", "Futurea LT"))) 
ggsave("figures/mass_spec_ratio.svg", p1)
#-------------------------------------------------------------------------------
odonates <- drop_na(odonates, abundance)

p2 <- ggplot(aes(x = abundance, y = spec_ratio), data = odonates) +
        geom_point() +
        geom_smooth(method = "gam", formula = y~x) +
        xlab("Abundnace of Host") +
        ylab("") +
        theme(text = element_text(family = c("Futura Medium", "Futurea LT")))
ggsave("figures/abun_spec_ratio.svg", p2 + ylab("Ratio of Specialists to Generalists"))

plot_grid(p1, p2)
```
<p class = "caption">
Figure 3.2 **Left panel:** The ratio of specialists to generalist on a host 
species versus the average mass of that species. **Right panel:** The ratio if 
specialists to generalist on a host species versus the abundance of that 
species. Mass is on a log scale. A single species was filtered out, 
*Leucorrhinia hudsonica*, which had only a single specialist and no generalists
recorded.
</p>

```{r}
source("scripts/specialization_gam_models.r")
# Checking the phylo model for sensitivity.
spec_gen_25 <- cbind(odonates$specialist_25, odonates$generalist_25)
phylo_model_25 <- gam(spec_gen_25 ~ s(species, bs = 'mrf', 
                                  xt = list(penalty = pmatrix), k = 10), 
                   data = odonates, method = "REML", family = binomial)
phylo_25_p <- summary(phylo_model_25)$p.table[,"Pr(>|z|)"]

spec_gen_50 <- cbind(odonates$specialist_50, odonates$generalist_50)
phylo_model_50 <- gam(spec_gen_50 ~ s(species, bs = 'mrf', 
                                  xt = list(penalty = pmatrix), k = 10), 
                   data = odonates, method = "REML", family = binomial)
phylo_50_p <- summary(phylo_model_25)$p.table[,"Pr(>|z|)"]

# Checking the mass model for sensitivity.
mass_model_25 <- gam(spec_gen_25 ~ log(mass), 
         data = odonates, method = "REML", family = binomial)
mass_25_p <- summary(mass_model_25)$p.table[2,"Pr(>|z|)"]

mass_model_50 <- gam(spec_gen_50 ~ log(mass), 
         data = odonates, method = "REML", family = binomial)
mass_50_p <- summary(mass_model_50)$p.table[2,"Pr(>|z|)"]

# Checking the abundance model for sensitivity.
spec_gen_abun_25 <- cbind(odonates_abun$specialist_25, 
                          odonates_abun$generalist_25)

abun_model_25 <- gam(spec_gen_abun_25 ~ log(abundance), 
                     data = odonates_abun, method = "REML", family = binomial)
abun_25_p <- summary(abun_model_25)$p.table[2,"Pr(>|z|)"]

spec_gen_abun_50 <- cbind(odonates_abun$specialist_50, 
                          odonates_abun$generalist_50)
abun_model_50 <- gam(spec_gen_abun_50 ~ log(abundance), 
                     data = odonates_abun, method = "REML", family = binomial)
abun_50_p <- summary(abun_model_50)$p.table[2,"Pr(>|z|)"]

```


|Model         |Cut-off|p-value          |
|:-------------|:-----:|:---------------:|
|Phylogeny     |50 Ma  |$`r phylo_50_p`$ |
|log(Abundance)|50 Ma  |$`r abun_50_p`$  |
|log(Mass)     |50 Ma  |$`r mass_50_p`$  |
|Phylogeny     |25 Ma  |$`r phylo_25_p`$ |
|log(Abundance)|25 Ma  |$`r abun_25_p`$  |
|log(Mass)     |25 Ma  |$`r mass_25_p`$  |
<p class = "caption">
Table 3.4 Results of sensitivity analysis for the cut-off value defining
generalist and specialist mites.
</p>

# 4	Discussion 

As expected from their higher parasitism intensity, we find more specialist 
parasites on damselflies than on dragonflies [@pinheiro2016]. We also found that
host abundance was a significant predictor of the ratio of specialists to
generalists on a host, but was a much weaker predictor than mass. In fact, mass
was almost exactly as strong of a predictor of specialization as the phylogeny
was (R^2^_adj_ = `r ms$r.sq` vs. `r ps$r.sq`) suggesting that mass might be the
trait which drives the phylogenetic trend. Mass was included in this analysis
since we have much more mass data and it was thought to be related to host
abundance. The strong predictive power seems to indicate that mass is also
strongly associated with other traits beyond abundance which drive selection
for or against parasite specialization on a host.

[@moller2005] saw a triangular pattern similar to ours where some lineages had
both generalists and specialists and some had only generalists in a bird-flea
system. They saw the specialists occurring only on the birds with stronger
immune responses. They're explanation was that antagonistic coevolution leads
the birds to develop stronger immune responses to parasites while the parasites
must become ever more specialized on a narrow set of hosts to overcome these
defences. In their study system, birds which had strong immune responses were
also colonial nesters. It might be that this process of antagonistic coevolution
only exerted extra selective pressure towards specialization on parasites which
already were specialist due to colonial birds being more reliable hosts. In a 
study of malaria parasites, the same association between host mass, host 
abundance, and parasite specialization was found, with the authors making the
argument of host reliability leads to specialization [@svensson-coelho2016]

Since the ecology of the system that @moller2005 studied is different from our
own in that strong host defences and high host reliability do not coincide in
the same lineage of hosts, we might expect to find the opposite relationship
between host defences and host specialization in our system. Dragonflies have
have thicker cuticles and a stronger immune response; they are also less 
attractive hosts since they are less abundant [@blondeau]. The extra predictive
power that mass had which abundance did not may be related to host defences. 
Smaller hosts may have less energy to expend on a strong immune response and a 
thick cuticle, producing an association between parasite specialization and 
mass.

Like many ecological studies, ours is subject to uneven and incomplete sampling.
Dragonflies are often much more difficult to catch than damselflies and so may
be less represented in our dataset that they are be in nature. Moreover we have
better coverage of the phylogeny where they are more abundant and easier to 
catch. More species in Coenagrionidae than in Gomphidae or Aeshnidae, etc.
It is also difficult to confidently declare a lack of interactions of parasite
interactions for a host species, especially when we have only caught that
species a few times and it is known to have low parasite intensity. Nonetheless
these results are unlikely to to be the result of any sampling artifact. 
the phylogenetic Simpson's diversity of a parasite's host community should be
biased to lower values when that community is undersampled. Undersampling should
have made Dragonflies appear to have more specialist parasites than they did.
Despite this, we found the pattern that damselflies had more specialist
parasites.

# 5 References

